src/ast.scm

```diff
@@ -310,6 +318,9 @@
 (define (kwarg? e)
   (and (pair? e) (eq? (car e) 'kw)))
 
+(define (nospecialize-meta? e)
+  (and (length> e 2) (eq? (car e) 'meta) (eq? (cadr e) 'nospecialize)))
```

Comment:
add another parameter `one::Bool` and check `(if one (eq? (length e) 3) (length> e 2))`